src/modules/75-cd/components/PipelineSteps/TerraformPlan/TerraformPlan.tsx

```diff
@@ -391,41 +539,68 @@ function TerraformPlanWidget(
                 />
               </Accordion>
             </>
-
-            {showModal && (
+            {showRemoteWizard && (
               <Dialog
-                onClose={() => setShowModal(false)}
-                enforceFocus={false}
-                className={cx(Classes.DIALOG, 'padded-dialog')}
-                {...modalProps}
-                title={getString('pipelineSteps.configFiles')}
+                {...DIALOG_PROPS}
+                isOpen={true}
                 isCloseButtonShown
-                style={{ padding: '24px' }}
+                onClose={() => {
+                  setConnectorView(false)
+                  setShowRemoteWizard(false)
+                }}
+                className={cx(css.modal, Classes.DIALOG)}
               >
-                <ConfigForm
-                  onClick={data => {
-                    const configObject = {
-                      ...data.spec?.configuration?.configFiles
-                    }
-
-                    if (configObject?.store.spec.gitFetchType === 'Branch') {
-                      delete configObject.store.spec.commitId
-                    } else if (configObject?.store.spec.gitFetchType === 'Commit') {
-                      delete configObject.store.spec.branch
-                    }
-
-                    const valObj = cloneDeep(formik.values)
-                    configObject.store.type = configObject?.store?.spec?.connectorRef?.connector?.type || 'Git'
-                    set(valObj, 'spec.configuration.configFiles', { ...configObject })
-
-                    formik.setValues(valObj)
-
-                    setShowModal(false)
-                  }}
-                  data={formik.values}
-                  onHide={() => setShowModal(false)}
-                  isReadonly={readonly}
-                  allowableTypes={allowableTypes}
+                <div className={css.createTfWizard}>
+                  <StepWizard title={getTitle()} className={css.configWizard} onStepChange={onStepChange}>
+                    <TerraformConfigStepOne
+                      name={getString('cd.configFileStepOne')}
+                      data={formik.values}
+                      isTerraformPlan
+                      isReadonly={readonly}
+                      isEditMode={isEditMode}
+                      allowableTypes={allowableTypes}
+                      setConnectorView={setConnectorView}
+                      selectedConnector={selectedConnector}
+                      setSelectedConnector={setSelectedConnector}
+                    />
+                    {connectorView ? getNewConnectorSteps() : null}
+                    <TerraformConfigStepTwo
+                      name={getString('cd.configFileDetails')}
+                      isTerraformPlan
+                      isReadonly={readonly}
+                      allowableTypes={allowableTypes}
+                      onSubmitCallBack={(data: any, prevStepData: any) => {
```

Comment:
Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232256)
```suggestion
                        onSubmitCallBack={(data: unknown, prevStepData: any) => {
```